### DISCUSSION TITLE
Header theme is no longer applied after updating packages

### DISCUSSION THREAD

**Discussion by Bergmam** [cloudscape-design/components#2582] [Q&A] [answered]
Hi, 

I updated the following packages:

```
@cloudscape-design/components: 3.0.484 -> 3.0.718
@cloudscape-design/components-themeable: 3.0.500 -> 3.0.733
```

After doing so, the theming I had previously applied to the `header` visual context is now completely ignored. 

Before updating packages: 
<img width="239" alt="Screenshot 2024-08-19 at 5 16 24 PM" src="https://github.com/user-attachments/assets/990ba5b2-d42b-48b3-933b-db4d4914105c">

After updating packages:
<img width="206" alt="Screenshot 2024-08-19 at 5 16 37 PM" src="https://github.com/user-attachments/assets/c8619062-be17-44b1-aa91-a8d520e2b2f5">

My theme:

```
contexts: {
    header: {
      tokens: {
        colorBackgroundLayoutMain: brandStyles.secondaryBrand90,
        colorBackgroundHomeHeader: brandStyles.secondaryBrand90,
        colorTextHeadingDefault: brandStyles.tertiaryBrand100,
        colorTextLinkDefault: brandStyles.primaryBrand80,
        colorTextBreadcrumbCurrent: brandStyles.secondaryBrand50,

        colorTextLinkButtonNormalDefault: brandStyles.primaryBrand80,
        colorBorderButtonNormalDefault: brandStyles.primaryBrand80,
        colorBackgroundButtonPrimaryDefault: brandStyles.primaryBrand100,

        colorBackgroundNotificationRed: brandStyles.utilityDanger100,
        colorBackgroundNotificationGreen: brandStyles.utilitySuccess100,
        colorBackgroundNotificationYellow: brandStyles.primaryBrand100,
        colorBackgroundNotificationBlue: brandStyles.primaryBrand100,
      },
    },
  }
}
```
The main issue is the `colorBackgroundLayoutMain` token of the header context which previously had a color applied to it, but after updating the packages it seems like it is just reverting back to the top-level design token which in my app is set to white (`#FFFFFF`).  The font colours are also wrong.

Inspecting the header background with the wrong color gives this, and so it does not look like it is picking the token from the header visual context at all.

<img width="557" alt="Screenshot 2024-08-19 at 5 22 48 PM" src="https://github.com/user-attachments/assets/5e1e2583-634e-48f6-8b2d-4bed0da01665">

Context is applied correctly for alerts and flashbars.

Is this a bug or did some functionality change between the versions that I am unaware of?

**Comment by Bergmam** [comment]
After closer inspection, it looks like after updating the packages the parent element is missing the  `awsui-context-content-header` class.

Before update:

<img width="525" alt="Screenshot 2024-08-20 at 11 07 43 AM" src="https://github.com/user-attachments/assets/1e93cb48-ef64-4cc8-b026-8028d8e45e39">


After update:

<img width="451" alt="Screenshot 2024-08-20 at 11 07 51 AM" src="https://github.com/user-attachments/assets/323b4ac9-9102-4b39-94bb-31b0e94d2172">

**Reply by just-boris** [reply, marked as answer]
Thanks for the extra info, now we can tell what happened. The dark header feature was removed, therefore the theming for a non-existing feature is not supposed to work either.

Details here: https://github.com/cloudscape-design/components/discussions/2235
